readers/influxdb/messages_test.go

```diff
@@ -72,14 +74,13 @@ func TestReadAll(t *testing.T) {
 	stringMsgs := []senml.Message{}
 	dataMsgs := []senml.Message{}
 	queryMsgs := []senml.Message{}
-	rand.Seed(time.Now().UnixNano())
-	to := msgsNum
-	now := float64(rand.Intn(to) + offset)
+	now := time.Now().Unix() * 1e9
 
 	for i := 0; i < msgsNum; i++ {
 		// Mix possible values as well as value sum.
 		msg := m
-		msg.Time = now - float64(i)
+		// now is unixnano, now/1e9 is seconds
```

Comment:
Why don't you set `now` immediately rather than multiplying and then dividing? Like:
```go
now := float64(time.Now().Unix())
for i := 0; i < msgsNum; i++ {
// Mix possible values as well as value sum.
msg := m
msg.Time = now - messageDelay*float64(i)
```